lucetc/src/function.rs

```diff
@@ -259,3 +266,94 @@ impl<'a> FuncEnvironment for FuncInfo<'a> {
         Ok(*pos.func.dfg.inst_results(inst).first().unwrap())
     }
 }
+
```

Comment:
My only substantive comment is that these fns could probably live in another file (func_manifest.rs) because the rest of this file is dedicated to implementing the FuncEnvironment trait used by cranelift-wasm. The two concepts are pretty disjoint.